Add message_thread_id for Telegram contact points

th0th opened the pull request:
See https://github.com/prometheus/alertmanager/pull/3638.

th0th committed:
Add message_thread_id for Telegram contact points

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/grafana/terraform-provider-grafana?pullRequest=1234) <br/>All committers have signed the CLA.

julienduchesne commented on internal/resources/grafana/resource_alerting_contact_point_notifiers.go:
```diff
@@ -1701,6 +1707,7 @@ func (t telegramNotifier) unpack(raw interface{}, name string) gapi.ContactPoint
 
 	unpackNotifierStringField(&json, &settings, "token", "bottoken")
 	unpackNotifierStringField(&json, &settings, "chat_id", "chatid")
+	unpackNotifierStringField(&json, &settings, "message_thread_id", "messagehtreadid")
```
```suggestion
	unpackNotifierStringField(&json, &settings, "message_thread_id", "messagethreadid")
```

th0th replied:
Nice catch 😅

julienduchesne commented on internal/resources/grafana/resource_alerting_contact_point_test.go:
```diff
@@ -264,6 +264,7 @@ func TestAccContactPoint_notifiers(t *testing.T) {
 					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.#", "1"),
 					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.0.token", "token"),
 					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.0.chat_id", "chat-id"),
+					resource.TestCheckResourceAttr("grafana_contact_point.receiver_types", "telegram.0.message_thread_id", "message-thread-id"),
```
hmm wondering why that passes. That's not the value in the test file

julienduchesne replied:
Oh wait, it doesn't. Tests hadn't run 😄

th0th committed:
Update resource_alerting_contact_point_notifiers.go

th0th commented:
I synced my branch with upstream 👍

julienduchesne commented:
Can't really merge this until it's supported in Grafana unfortunately

vanpt1114 commented:
I think Grafana Alerting has supported the Telegram message_thread_id, I can observe it from the UI, so is this PR good to merge?

julienduchesne committed:
Fix casing and tests
The attribute is `message_thread_id`, according to https://github.com/grafana/grafana/pull/79546

julienduchesne committed:
Try with an integer
